.github/workflows/openapi-compatibility-tests.yaml

```
@@ -0,0 +1,40 @@
+name: lakeFS OpenAPI Compatibility Tests
+
+on:
+  pull_request:
+    paths:
+      - "api/**"
+  push:
+    branches:
+      - master
+  workflow_dispatch: {}
+
+# These permissions are needed to interact with GitHub's OIDC Token endpoint.
+permissions:
+  id-token: write
+  contents: read
+
+jobs:
+  swagger-backward-compatibility:
+    name: Run Swagger compatibility tests
+    runs-on: ubuntu-20.04
+    steps:
+        - name: Check-out revision code
+          uses: actions/checkout@v4
```

Consider checking this out into
```suggestion
          uses: actions/checkout@v4
          with:
            path: 'changed'
```
to be easier to read.  (And also @nopcoder's changes of course...)